cfme/markers/upgrade.py

```diff
@@ -0,0 +1,40 @@
+"""Marker definitions for upgrade testing"""
+from cfme.fixtures.pytest_store import store
+
+
+# Add Marker
+def pytest_configure(config):
+    config.addinivalue_line("markers", "post_upgrade: Mark test for post upgrade testing")
+
+
+# Filtering options
+def pytest_addoption(parser):
+    group = parser.getgroup("cfme")
+    group.addoption(
+        "--upgrade-appliance",
+        dest="upgrade_appliance",
+        action="store_true",
+        default=False,
+        help="Upgrade an appliance before tests are run.",
+    )
+    group.addoption(
+        "--upgrade-to",
+        dest="upgrade_to",
+        action="store",
+        default="5.11.z",
+        help="Supported versions 5.9.z, 5.10.z, 5.11.z (.z means latest and default is 5.11.z)",
+    )
+
+
+def pytest_sessionstart(session):
+    if store.parallelizer_role == "master":
```

Comment:
we have to get rid of session scoped fixtures

Reply:
I need to discuss with you... as per me we need session scope as we can not upgrade appliance for each test even not consider `module` scope as we are targeting multiple test from multiple file.